editor/editor_file_system.cpp

```diff
@@ -1816,13 +1822,19 @@ void EditorFileSystem::update_files(const Vector<String> &p_script_paths) {
 			if (fs->files[cpos]->type == SNAME("PackedScene")) {
 				_queue_update_scene_groups(file);
 			}
+			if (fs->files[cpos]->type == SNAME("Resource") || old_script_class_icon_path != fs->files[cpos]->script_class_icon_path) {
```

Comment:
The first check returns true only for plain Resources. Is this intended?

Reply:
Yes it's intended, the purpose here is to manage the modification of the script on a Resource here:

![image](https://github.com/godotengine/godot/assets/81109165/ac231ed7-c7d0-4b33-bc96-ca0c550bfd42)

The type is "Resource":
![image](https://github.com/godotengine/godot/assets/81109165/79ec53db-91ef-446e-97c4-8aa05297180a)

Reply:
Yeah but *technically* you can make a custom sub-class of another Resource type, e.g. Texture2D. But maybe it's not worth worrying about 🤔

Reply:
You are right!! Maybe I should use `ClassDB::is_parent_class` ??

Reply:
I wonder. Technically any file is a Resource sub-class, so maybe it's better to leave it like that for now.

Reply:
You are right, even a PackedScene is a Resource. I will leave it like that.